pkg/istio-agent/xds_proxy.go

```diff
@@ -367,7 +369,11 @@ func (p *XdsProxy) handleStream(downstream adsStream) error {
 
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
 	defer cancel()
-	upstreamConn, err := grpc.DialContext(ctx, p.istiodAddress, p.istiodDialOptions...)
+	upstreamAddr := p.istiodAddress
+	if p.istiodAddressOverride != "" {
+		upstreamAddr = p.istiodAddressOverride
+	}
+	upstreamConn, err := grpc.DialContext(ctx, upstreamAddr, p.istiodDialOptions...)
```

Comment:
the proxyLog.Errorf() below is still using the `p.istiodAddress`

Reply:
Will fix.

Reply:
Reverted to use discovery address, log is good now. If the error is related to validating the cert - will be logged in the err message.